### PR TITLE
cool#10630 doc electronic sign: get the signature on the server

### PR DESCRIPTION
**Pull request by vmiklos**
- **cool#10630 doc electronic sign: get the signature on the server**
- **cool#10630 doc electronic sign: stop sending the secret to the client**
- **cool#10630 doc electronic sign: remove CSP rules**

### CODE REVIEW

**Comment by vmiklos** [comment]
Rebase to be on top of latest master + switch to non-throwing when the 'body' produced by JS is not there.

File: kit/Kit.cpp [hunk 1]
```diff
@@ -1711,23 +1711,20 @@ void Document::notifyViewInfo()
...
                         Poco::Dynamic::Var var = parser.parse(userPrivateInfo);
                         Object::Ptr userPrivateInfoObj = var.extract<Object::Ptr>();
-                        std::string signatureCert;
-                        JsonUtil::findJSONValue(userPrivateInfoObj, "SignatureCert", signatureCert);
-                        if (!signatureCert.empty())
+                        std::initializer_list<std::string> keys = {
+                            "SignatureCert",
+                            "SignatureKey",
+                            "SignatureCa",
+                            "ESignatureSecret",
```
Comment (vmiklos) [inline kit/Kit.cpp:1718]: This is just adding the ESignatureSecret as a key to be filtered + refactor to avoid further copy & paste.

**Comment by vmiklos** [comment]
@caolanm could you please review this? Thanks.

The second half of this "move secret to the server" story. I got this wrong initially as following Zotero as an example looked like a good idea; but eideasy is more nuanced with its split clientid + secret token pair.

**Review by caolanm — APPROVED** [review]
looks sane to me

**Pull request merged**